### PR TITLE
MOM standard proposal

### PR DESCRIPTION
**Pull request by Neurone**

### CODE REVIEW

File: EIPS/eip-draft_my_own_messages.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+---
+eip: <to be assigned>
```
Comment (MicahZoltu) [inline EIPS/eip-draft_my_own_messages.md:2]: Use `eip-2848` and make sure to update the EIP file name as well.

File: EIPS/eip-draft_my_own_messages.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+title: My Own Messages (MOM)
+author: Giuseppe Bertone (@Neurone)
+discussions-to: https://github.com/InternetOfPeers/EIPs/issues/1
+status: Draft
+type: Standards Track
+category (*only required for Standard Track): ERC
```
Comment (MicahZoltu) [inline EIPS/eip-draft_my_own_messages.md:8]:
```suggestion
category: ERC
```

File: EIPS/eip-draft_my_own_messages.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+My Own Messages (MOM) is a standard to notarize and share messages using Ethereum, so that contents are verificable and decentralized.
+
+Main advantages:
+- You can send messages to users of your ÐApp or Smart Contract, and they always know it is a voice reliable as the smart contract is.
+- Create your Ethereum account dedicated to your personal messages, say something only once and it can be seen on every social platform (no more reply of the same post/opinion on dozens of sites like Reddit, Twitter, Facebook, Medium, Disqus, and so on...)
+- Small fee to be free: pay just few cents of dollar to notarize your messages, and distribute them with IPFS, Swarm or any other storage you prefer. Because the multihash of the content is notarized, you can always check the integrity of the message you download even from centralized storage services.
+- Finally, you can ask and get tips for your words directly into your wallet.
+
+I know, My Own Messages (MOM) sounds like _mom_. And yes, pun intended :)
```
Comment (MicahZoltu) [inline EIPS/eip-draft_my_own_messages.md:24]: The simple summary should be about a sentence.  Most of this should probably go in the motivation section, which is the appropriate place to discuss *why* this standard exists.

File: EIPS/eip-draft_my_own_messages.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+My Own Messages (MOM) is a standard to notarize and share messages using Ethereum. You can give voice to your smart contracts, send messages to the world, create a certified blog with your ideas, and so on.
```
Comment (MicahZoltu) [inline EIPS/eip-draft_my_own_messages.md:28]: This section should include a bit more technical detail about how you achieve that.  You don't want to rewrite the specification here, but give a very high level overview of what this standard achieves and how it achieves it.  Something like "...this is achieved by submitting a self-send message to the blockchain with a specific format and message data attached.

File: EIPS/eip-draft_my_own_messages.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+## Valid MOM transaction's structure
+
+| ATTRIBUTE | VALUE |
+|:--------|:------------|
+| `from` | `MUST` be the tx signer |
+| `to` | `MUST` be the tx signer |
+| `value` |  `MUST` be 0 wei |
+| `data` | `MUST` be at least 1 byte. First byte is the operational code, then it comes the content. |
```
Comment (MicahZoltu) [inline EIPS/eip-draft_my_own_messages.md:49]: Recommend using RFC 2119 style here (and consider adding the RFC 2119 blurb at the top of the Specification section).

**Comment by Neurone** [comment]
Thanks for the feedbacks. I'm doing the changes you suggested and I'll push an updated commit soon.

File: EIPS/eip-2848.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+As a _developer_ or _pool's owner_, I'd like to send messages to my users in a decentralized way. They must be able to easily verify my role in the smart contract context (owner, user, and so on) and they must be able to do it without relying on external, insecure and hackable social media sites (Facebook, Twitter, you name it). Also, I'd like to read messages from my userbase, in the same secure and verifiable manner.
+
+As a _user_, I want a method to easily share my thoughts and idea, publish content, send messages, receive feedbaks, receive tips, and so on, wihout dealing with any complexity: just write a message, send it and it's done. Also, I want to write to some smart contract's ower or to the sender of some transaction.
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:26]:
```suggestion
As a _user_, I want a method to easily share my thoughts and idea, publish content, send messages, receive feedbaks, receive tips, and so on, without dealing with any complexity: just write a message, send it and it's done. Also, I want to write to some smart contract's owner or to the sender of some transaction.
```

File: EIPS/eip-2848.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+Please also note that, because it's usually up to the author of a message to be sure the content is available to everyone, if a `DELETE` message was sent it's very likely the content referenced by the multihash isn't available anymore, simply because probably it's not shared by anyone.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+Ethereum is _account based_, so it's good to be identified as a single source of information.
+
+It is also able of doing notarization very well and to impose some restrictions on transaction's structure, so it's good for commands.
+
+IPFS, Swarm or other CANs (Content Addressable Networks) or storage methods are good to store lot of informations. So, the union of both worlds it's a good solution to achieve the objectives of this message standard.
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:102]:
```suggestion
IPFS, Swarm or other CANs (Content Addressable Networks) or storage methods are good to store a lot of information. So, the union of both worlds it's a good solution to achieve the objectives of this message standard.
```

File: EIPS/eip-2848.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+| ATTRIBUTE | VALUE |
+|:--------|:------------|
+| `to` | **MUST** be the same account signing the transaction |
+| `value` |  **MUST** be `0` wei |
+| `data` | **MUST** be at least `1` byte. The first byte **MUST** be operational code and following bytes **MUST** be based on the operational codes listed below. |
+
+#### List of supported operations and messages
+Each operational code has one or more parameters, and all parameters **MUST** be considered mandatory.
+
+Optional parameters don't exist: if parameters for the specific operational code are not all present or they don't follow the rules, clients **MUST** ignore the transaction completly.
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:73]:
```suggestion
Optional parameters don't exist: if parameters for the specific operational code are not all present or they don't follow the rules, clients **MUST** ignore the transaction completely.
```

File: EIPS/eip-2848.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+As a _developer_ or _pool's owner_, I'd like to send messages to my users in a decentralized way. They must be able to easily verify my role in the smart contract context (owner, user, and so on) and they must be able to do it without relying on external, insecure and hackable social media sites (Facebook, Twitter, you name it). Also, I'd like to read messages from my userbase, in the same secure and verifiable manner.
+
+As a _user_, I want a method to easily share my thoughts and idea, publish content, send messages, receive feedbaks, receive tips, and so on, wihout dealing with any complexity: just write a message, send it and it's done. Also, I want to write to some smart contract's ower or to the sender of some transaction.
+
+As an _explorer service_, I want to give my users an effective way to read informations by smart contract owners and a place to share ideas and informations, without using third party services (i.e. Etherscan uses Disqus, and so on)
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:28]:
```suggestion
As an _explorer service_, I want to give my users an effective way to read information by smart contract owners and a place to share ideas and information without using third party services (i.e. Etherscan uses Disqus, and so on)
```

File: EIPS/eip-2848.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+MOM wants to be useful, easy to implement and read, error proof, fast and cheap, but:
+- using a smart contract for messages can leads more easily to errors and misunderstandings:
+    - address of the contract can be wrong
+    - smart contract must be deployed on that specific network to send messages
+- executing a smart contract costs much more then sending transactions
+- executing a smart contract just to store static data is the best example of an anti-pattern (expensive and almost useless)
+
+Without a specific smart contract to rely on, the MOM standard can be implemented and used right now in any existing networks, and even in future ones.
+
+Finally, if you can achive exactly the same result without a smart contract, you didn't need a smart contract at the first place.
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:116]:
```suggestion
Finally, if you can achieve exactly the same result without a smart contract, you didn't need a smart contract at the first place.
```

File: EIPS/eip-2848.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+There's no benefit to store _static_ messages on-chain, if they are not related to some smart contract's state or if they don't represent exchange of value. The cost of storing data on-chain is also very high.
+
+### Why not storing op codes inside the message?
+While cost effectiveness is a very important feature in a blockchain related standard, there's also a compromise to reach with usability and usefulness.
+
+Storing commands inside the messages forces the client to actually download messages to understand what to do with them. This is very unefficent, bandwidth and time consuming.
+
+Being able to see the commands before downloading the content, it allows the client to recreate the history of all messages and then, at the end, download only updated messages.
+
+Creating a structure for the content of the messages leads to many issues and considerations in parsing the contect, if it's correct, mispelled, and so on.
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:128]:
```suggestion
Creating a structure for the content of the messages leads to many issues and considerations in parsing the content, if it's correct, misspelled, and so on.
```

File: EIPS/eip-2848.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+
+## Security Considerations
+<!--All EIPs must contain a section that discusses the security implications/considerations relevant to the proposed change. Include information that might be important for security discussions, surfaces risks and can be used throughout the life cycle of the proposal. E.g. include security-relevant design decisions, concerns, important discussions, implementation-specific guidance and pitfalls, an outline of threats and risks and how they are being addressed. EIP submissions missing the "Security Considerations" section will be rejected. An EIP cannot proceed to status "Final" without a Security Considerations discussion deemed sufficient by the reviewers.-->
+MOM is very simple and it has no real security concerns by itself. The standard already considers valid only transactions with `0` value and where `from` and `to` addresses are equals.
+
+The only concerns can come from the payload, but it is more realted to the client and not to the standard itself, so here you can find some security suggestions related to clients implementing the standard.
```
Comment (MicahZoltu) [inline EIPS/eip-2848.md:162]:
```suggestion
The only concerns can come from the payload, but it is more related to the client and not to the standard itself, so here you can find some security suggestions related to clients implementing the standard.
```

**Comment by Neurone** [comment]
> The automated spellchecker is complaining. I think I have left suggestions to fix them all.

Many thanks! I accepted your fixes and I found some other typos, sorry for that ^^. I'll submit a new update soon.

**Comment by Neurone** [comment]
Done: #2855 Thanks @MicahZoltu for your work and patience!

**Pull request merged**